specs/Reservations-specs.rb

```diff
@@ -0,0 +1,232 @@
+    end
+    after do
+      @new_hotel.clear_reservations
+    end
+    it 'must return an array' do
+      @new_hotel.list_rooms_available_by_date("2018-01-02").must_be_kind_of Array
+    end
+    it 'must return the correct number of rooms available' do
+      @new_hotel.list_rooms_available_by_date("2018-01-03").length.must_equal 17
+    end
+    it 'must return the correct rooms available' do
+      @new_hotel.list_rooms_available_by_date("2018-01-03")[0].room_number.must_equal 3
+    end
+  end
+  # describe 'assign room number' do
+  #   it 'will assign a room number if no room number is given' do
+  #     new_booking = @new_hotel.new_reservation("2017-09-21", "2017-09-23")
+  #     new_booking.room_number.wont_equal 0
+  #   end
+  # end
+
+  # describe 'validate room number' do
+  #   after do
+  #     @new_hotel.clear_reservations
+  #   end
+  #   it 'must raise an error if a room number that does not exist is entered' do
+  #     proc{new_booking = @new_hotel.new_reservation("2017-09-21", "2017-09-30", 45)}.must_raise ArgumentError
+  #   end
+  # end
+
+  describe 'all reservations' do
+    before do
+      @new_hotel.clear_reservations
+      @new_booking1 = @new_hotel.new_reservation("2017-09-21", "2017-09-23", 2)
+      @new_booking2 = @new_hotel.new_reservation("2020-01-01", "2020-01-15", 3)
+      @new_booking3 = @new_hotel.new_reservation("2019-01-01", "2019-01-15", 5)
+    end
+    it 'must be an array of all reservations' do
+      @new_hotel.all_reservations.must_be_kind_of Array
+    end
+    it 'must contain correct number of reservations made' do
+      @new_hotel.all_reservations.length.must_equal 3
+    end
+    it 'must contain the first reservation made' do
+      @new_hotel.all_reservations[0].check_in.must_equal Date.parse("2017-09-21")
+    end
+    it 'must contain the correct total cost of the reservation' do
+      @new_hotel.all_reservations[0].total_cost.must_equal 400
+    end
+  end
+
+  describe 'list_reservations_by_date' do
+    before do
+      @new_hotel.clear_reservations
+      @new_booking1 = @new_hotel.new_reservation("2017-09-21", "2017-09-23", 1)
+      @new_booking2 = @new_hotel.new_reservation("2019-01-01", "2019-01-15", 10)
+      @new_booking3 = @new_hotel.new_reservation("2019-01-02", "2019-01-12", 12)
+      @new_booking4 = @new_hotel.new_reservation("2019-01-01", "2019-01-13", 15)
+    end
+    it "must be a method of Reservations" do
+      @new_hotel.must_respond_to :list_reservations_by_date
+    end
+    it "must be an array" do
+      @new_hotel.list_reservations_by_date("2019-01-01").must_be_kind_of Array
+    end
+    it "must have the correct number of reservations for the date given" do
+      @new_hotel.list_reservations_by_date("2019-01-04").length.must_equal 3
+    end
+    it "must return the correct reservations" do
+      @new_hotel.list_reservations_by_date("2017-09-21")[0].room_number.must_equal @new_booking1.room_number
+    end
```

Comment:
You're missing some test cases here - they look very similar to the ones I would like to see for listing rooms by date.